bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -951,8 +1038,92 @@ private Comparator<Voice> createVoiceComparator(Locale locale) {
         return null;
     }
 
+    private void stopDialogs(Predicate<DialogProcessor> filter) {
+        synchronized (dialogRegistrationStorage) {
+            var dialogsToStop = dialogProcessors.values().stream().filter(filter).toList();
+            if (dialogsToStop.isEmpty()) {
+                return;
+            }
+            for (var dialog : dialogsToStop) {
+                stopDialog(dialog.dialogContext.source());
+            }
+        }
+    }
+
+    private void debounceBuildDialogRegistrations() {
```

Comment:
Why is this needed?

Reply:
It's not really needed, but I think it will skip a lot of unnecessary executions, as this needs to be called on every related add-on registration but there is not need to do it immediately, I added it thinking in the following cases:

* On the initial start of openHAB this method will be called a lot of times.
* Normally the sink and source related to the dialog will bellow to the same add-on so registration will happen mostly at same time.

Let me know if you think is better to remove/improve it.

Reply:
Understood. Looks useful. Maybe it's better to call it `scheduleDialogRegistrations` and add JavaDoc explaining that each call re-schedules the build.